zyte_common_items/items.py

```diff
@@ -712,3 +713,38 @@ class RealEstate(Item):
 
     #: Contains metadata about the data extraction process.
     metadata: Optional[Metadata] = None
+
+
+@attrs.define(kw_only=True)
+class ProductNavigation(Item):
+    """Represents navigations for a page of product listings on a e-commerce website"""
+
+    #: Main URL from which the data is extracted.
+    url: str = attrs.field(converter=url_to_str)
+
+    #: Name of the category/page with the product list.
+    #:
+    #: Format:
+    #:
+    #: - trimmed (no whitespace at the beginning or the end of the description string),
+    categoryName: Optional[str] = None
+
+    #: List of sub-category links ordered by their position in the page.
+    subCategories: Optional[List[Navigation]] = None
+
+    #: List of product links found on the page category ordered by their position in the page.
+    items: Optional[List[Navigation]] = None
+
+    #: The navigation to the next page, if available.
+    nextPage: Optional[Navigation] = None
```

Comment:
If I'm reading the schema correctly, nextPage's Navigation is different from other Navigation types, because it doesn't have metadata. So, if we want to be precise, we need to use another Navigation class for the nextPage link. I'd probably prefer to address it in the schema definition though, not here.

Reply:
>  I'd probably prefer to address it in the schema definition though, not here.

@kmike do you mean to change the schema to include probability for `nextPage`?

Reply:
@kmike to be clear, are you discussing how to best organize the common items definitions, or how to change the schema to simplify it? In other words, do we have a constraint that they need to be exactly the same, or are you considering simplifications in commit items which don't have to be in the schema? I guess there would be an obvious tension here, where we don't want to expose fields we don't need in the schemas, which leads to a simpler single schema, but results in less re-use of code when you define multiple schemas.

Reply:
What are the action points here? Do I need to change anything on the PR ?

Reply:
Hey @VMRuiz! As the schema is not changed for now (no decision), we should change the item definition to follow the schema more precisely, i.e. no metadata for nextPage link.

Reply:
Is https://github.com/zytedata/zyte-common-items/pull/37/commits/d688ff783c649fcd69ed1c18945bc0a7dd4cdb88 a good solution for now?

Reply:
@Gallaecio I think this solution is quite precise; it has an advantage of representing the schema correctly. The disadvantage is the weird developer API, when you need to be using ProbabilityRequest objects, which make very little sense for the use case of writing the extraction code.

What do you think about adding a default processor for the base page object classes, to allow users to return Request objects everywhere, and "promoting" them to ProbabilityRequest in the processor?

Reply:
We may also be able to solve some of the issues with Metadata this way, e.g. by allowing users to return Metadata, and "demoting" it to classes with less fields if needed. In this case we can also add searchText to Metadata.

Reply:
So, to summarize:

* For the developers we provide a single Metadata and a single Request class.
* Then, depending on the data type, we convert them to more precise types. We can do it in default processors. We may also duplicate it in attrs convertors, so that it works without processors as well.

What do you think?